flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGateTest.java

```diff
@@ -499,20 +504,23 @@ public void testUpdateUnknownInputChannel() throws Exception {
 			assertThat(inputGate.getInputChannels().get(localResultPartitionId.getPartitionId()),
 				is(instanceOf((UnknownInputChannel.class))));
 
+			ResourceID localLocation = ResourceID.generate();
+
 			// Trigger updates to remote input channel from unknown input channel
-			inputGate.updateInputChannel(new InputChannelDeploymentDescriptor(
-				remoteResultPartitionId,
-				ResultPartitionLocation.createRemote(InputChannelBuilder.STUB_CONNECTION_ID)));
+			ShuffleDescriptor remoteSdd = createRemoteWithIdAndLocation(
+				remoteResultPartitionId.getPartitionId(),
+				ResourceID.generate());
+			inputGate.updateInputChannel(localLocation, new PartitionInfo(new IntermediateDataSetID(), remoteSdd));
 
 			assertThat(inputGate.getInputChannels().get(remoteResultPartitionId.getPartitionId()),
 				is(instanceOf((RemoteInputChannel.class))));
 			assertThat(inputGate.getInputChannels().get(localResultPartitionId.getPartitionId()),
 				is(instanceOf((UnknownInputChannel.class))));
 
 			// Trigger updates to local input channel from unknown input channel
-			inputGate.updateInputChannel(new InputChannelDeploymentDescriptor(
-				localResultPartitionId,
-				ResultPartitionLocation.createLocal()));
+			ShuffleDescriptor localSdd = createRemoteWithIdAndLocation(
+				localResultPartitionId.getPartitionId(), localLocation);
+			inputGate.updateInputChannel(localLocation, new PartitionInfo(new IntermediateDataSetID(), localSdd));
```

Comment:
could be reuse `createPartitionInfo(localResultPartitionId, localLocation)` to create `PartitionInfo` here.